src/services/expenses.service.js

```diff
@@ -0,0 +1,97 @@
+'use strict';
+
+const { filterExpenses } = require('../helpers/filterExpenses');
+const { Expense } = require('../models/Expense');
+
+async function getAllExpenses(userId, categories, from, to) {
+  const expensesFromDB = await Expense.findAll();
+
+  if (!userId && !categories && !from && !to) {
+    return expensesFromDB;
+  }
+
+  try {
+    const filteredExpenses = filterExpenses(
+      expensesFromDB, categories, userId, from, to
+    );
+
+    return filteredExpenses;
+  } catch (error) {
+    return error;
+  }
+}
```

Comment:
- Query from the database only expenses that you actually need instead of querying all and then filtering them.
https://sequelize.org/docs/v6/core-concepts/model-querying-basics/

- `try/catch` in this case is redundant, and it's a bad idea to return an error, as you try to catch it in the `getExpenses` function, but it's impossible if you just return it